Sources/ManageSmileys.php

```diff
@@ -1846,35 +1846,130 @@ function ImportSmileys($smileyPath)
 		fatal_lang_error('smiley_set_unable_to_import');
 
 	$allowedTypes = array('gif', 'png', 'jpg', 'jpeg', 'tiff', 'svg');
+	$known_sets = explode(',', $modSettings['smiley_sets_known']);
+	sort($known_sets);
 
+	// Get the smileys in the folder
 	$smileys = array();
 	$dir = dir($modSettings['smileys_dir'] . '/' . $smileyPath);
 	while ($entry = $dir->read())
 	{
 		$pathinfo = pathinfo($entry);
-		if (in_array($pathinfo['extension'], $allowedTypes) && $pathinfo['filename'] != 'blank')
-			$smileys[strtolower($entry)] = $entry;
+		if (in_array($pathinfo['extension'], $allowedTypes) && $pathinfo['filename'] != 'blank' && strlen($pathinfo['basename']) <= 48)
+			$smiley_files[strtolower($pathinfo['basename'])] = $pathinfo['basename'];
 	}
 	$dir->close();
 
-	// Exclude the smileys that are already in the database.
+	// Get the smileys that are already in the database.
+	$existing_smileys = array();
 	$request = $smcFunc['db_query']('', '
-		SELECT filename
-		FROM {db_prefix}smiley_files
-		WHERE filename IN ({array_string:smiley_list})
-			AND smiley_set = {string:smiley_set}',
+		SELECT *
```

Comment:
Why a star?
One of the db basic stuff,
don't use a star...
https://stackoverflow.com/a/3640051